tendermint/starport
# feat(`network`): simulate requests

- Add a routine `SimulateRequests` in `networkchain` to dynamically verify the requests by applying them to the current genesis and checking if genesis can be started
- Moved all request verification logic into `networktypes`
- Add a `starport network request verify` command to verify the request without taking any actions

As before, if we filter the `validator set nil` SDK error in the case no gentx exists and only accounts are added 

To test:
Spin up a local `spn` node or use `nightly` and test `starport n request verify` or `starport n request approve` commands

I tested manually the command with:
- Normal AddAccount + AddValidator requests
- Only AddAccount requests
- Wrong accounts
- Insufficient balance for the self-delegation
- Changing manually the signature of the gentx generated by the `init` command

The chain prepare steps are printed when verifying requests and the whole chain error is printed if the chain fails to start. There is some room for improvement for the console output, suggestions are welcomed
